main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala

```diff
@@ -34,28 +34,27 @@ object BoolUnification {
   /**
     * Returns the most general unifier of the two given Boolean formulas `tpe1` and `tpe2`.
     */
-  def unify(tpe1: Type, tpe2: Type, renv0: RigidityEnv)(implicit flix: Flix): Result[(Substitution, List[TypeConstraint]), UnificationError] = {
+  def unify(tpe1: Type, tpe2: Type, renv0: RigidityEnv)(implicit flix: Flix): Option[Substitution] = {
```

Comment:
Why do you not need a unification error anymore?

Reply:
Our results are now only "success" and "try again later". We can inspect kinds and provenance in unsolved constraints to give specific errors.